Code/GraphMol/StructChecker/Stereo.cpp

```diff
@@ -0,0 +1,864 @@
+  }
+  unsigned element = mol.getAtomWithIdx(iatom - 1)->getAtomicNum();
+  if (multiple && stereo && 15 != element && 16 != element) {  // "P" && "S"
+    if (allene)
+      return (ALLENE_PARITY);
+    else {
+      // stereo_error = "AtomParity: Stereobond at unsaturated atom";
+      std::cerr << "stereobond at unsaturated atom" << std::endl;
+      return (ILLEGAL_REPRESENTATION);
+    }
+  } else if (multiple && 16 != element)  // "S"
+    return (UNDEFINED_PARITY);
+
+  stereo_ligands[0].angle = 0.0; /* comp. angle rel. to first ligand */
+  for (unsigned i = 1; i < nbp.Atoms.size(); i++)
+    stereo_ligands[i].angle = Angle(stereo_ligands[0].x, stereo_ligands[0].y,
+                                    stereo_ligands[i].x, stereo_ligands[i].y);
+  for (unsigned i = 2; i < nbp.Atoms.size(); i++) /* sort ligands */
+    for (unsigned j = i; j > 1; j--)
+      if (stereo_ligands[j].angle < stereo_ligands[j - 1].angle) {
+        h = stereo_ligands[j];
+        stereo_ligands[j] = stereo_ligands[j - 1];
+        stereo_ligands[j - 1] = h;
+      } else
+        break;
+
+  return (nbp.Atoms.size() == 3 ? Atom3Parity(stereo_ligands)
+                                : Atom4Parity(stereo_ligands));
+}
+
+bool CheckStereo(const ROMol &mol) {
+  /*
+  * Checks if all potential stereocenters are either completely undefined
+  * or attributed with hashes and wedges according to MDL rules.
+  */
+  bool result = true;
+  int parity;
+  bool center_defined = false;
+  std::vector<Neighbourhood> neighbour_array;
+
+  SetupNeighbourhood(mol, neighbour_array);
+
+  for (unsigned i = 0; i < neighbour_array.size(); i++) {
+    const Neighbourhood &nbp = neighbour_array[i];
+    unsigned element = mol.getAtomWithIdx(i)->getAtomicNum();
+    unsigned n_ligands = (unsigned)nbp.Bonds.size();
+    if ((n_ligands > 2 && n_ligands <= 4) && (6 == element       // "C"
+                                              || 16 == element   // "S"
+                                              || 7 == element    // "N"
+                                              || 8 == element    // "O"
+                                              || 15 == element   // "P"
+                                              || 14 == element)  // "Si"
+        ) {
+      parity = AtomParity(mol, i + 1, nbp);
+      if (parity == ILLEGAL_REPRESENTATION) {  // stereo_error
+        result = false;
+      } else if (parity == EVEN_PARITY || parity == ODD_PARITY ||
+                 parity == ALLENE_PARITY)
+        center_defined = true;
+      else {
+        for (unsigned j = 0; j < nbp.Bonds.size(); j++) {
+          const Bond &bond = *mol.getBondWithIdx(j);
+          if (bond.getBeginAtomIdx() == i + 1 &&
+              (RDKit::Bond::STEREOZ == bond.getStereo()         // == UP
+               || RDKit::Bond::STEREOE == bond.getStereo())) {  // == DOWN))
+            // stereobond to non-stereogenic atom
+            std::cerr << "stereobond to nonstereogenic" << std::endl;
```

Comment:
It's more likely that the test should be using RDKit::Bond::BEGINWEDGE and RDKit::Bond::BEGINDASH